ui/src/status/reducers/JSONFeed.ts

```diff
@@ -1,19 +1,27 @@
-import * as actionTypes from 'src/status/constants/actionTypes'
+import {JSONFeedData} from 'src/types'
+import {Action, ActionTypes} from 'src/status/actions'
 
-const initialState = {
+export interface State {
+  hasCompletedFetchOnce: boolean
+  isFetching: boolean
+  isFailed: boolean
+  data: JSONFeedData
+}
+
+const initialState: State = {
   hasCompletedFetchOnce: false,
   isFetching: false,
   isFailed: false,
   data: null,
 }
 
-const JSONFeedReducer = (state = initialState, action) => {
+const JSONFeedReducer = (state: State = initialState, action: Action) => {
```

Comment:
what do you think about trying to type the output here? that would give us the JSONFeedReducerState type, or something to that effect, and it could be used wherever we destructure that state from redux, for instsance

Reply:
done